File: helpdeskspringboot/src/main/java/com/ez/payload/Last7DaysReportResponse.java
package com.ez.payload;

public interface Last7DaysReportResponse {

    // "dayMonth" column.
    // ex: Fri-21/4
    String getDayMonth();

    // "numOfNewTickets" column
    long getNumOfNewTickets();

    // "numOfSolvedTickets" column
    long getNumOfSolvedTickets();

    // "numOfClosedTickets" column
    long getNumOfClosedTickets();

    // "totalSpentHour" column
    float getTotalSpentHour();

}

File: helpdeskangular/src/app/payload/Last7DaysReportResponse.ts
export class Last7DaysReportResponse {

    constructor(

        // priority = id + name + resolveIn(hours)
        public dayMonth: string,

        // team = id + name + assignment method
        public numOfNewTickets: number,

        // number of tickets from fromDate to toDate of each [priorityid, teamid]
        public numOfSolvedTickets: number,

        // number of ontime tickets
        public numOfClosedTickets: number,

        // number of lated tickets
        public totalSpentHour: number



    ) {

    }

}
